huaweicloud/services/cfw/resource_huaweicloud_cfw_black_white_list.go

```diff
@@ -0,0 +1,332 @@
+		&createBlackWhiteListOpt)
+	if err != nil {
+		return diag.Errorf("error creating black white list: %s", err)
+	}
+
+	createBlackWhiteListRespBody, err := utils.FlattenResponse(createBlackWhiteListResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.id", createBlackWhiteListRespBody)
+	if err != nil {
+		return diag.Errorf("error creating black white list: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceBlackWhiteListRead(ctx, d, meta)
+}
+
+func buildCreateBlackWhiteListBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"object_id": utils.ValueIngoreEmpty(d.Get("object_id")),
+		"list_type": utils.ValueIngoreEmpty(d.Get("list_type")),
+		// direction can be 0
+		"direction": d.Get("direction"),
+		"protocol":  utils.ValueIngoreEmpty(d.Get("protocol")),
+		"port":      utils.ValueIngoreEmpty(d.Get("port")),
+		// address_type can be 0
+		"address_type": d.Get("address_type"),
+		"address":      utils.ValueIngoreEmpty(d.Get("address")),
+	}
+	return bodyParams
+}
+
+func resourceBlackWhiteListRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getBlackWhiteList: Query the CFW black white list detail
+	var (
+		getBlackWhiteListHttpUrl = "v1/{project_id}/black-white-lists"
+		getBlackWhiteListProduct = "cfw"
+	)
+	getBlackWhiteListClient, err := cfg.NewServiceClient(getBlackWhiteListProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	getBlackWhiteListPath := getBlackWhiteListClient.Endpoint + getBlackWhiteListHttpUrl
+	getBlackWhiteListPath = strings.ReplaceAll(getBlackWhiteListPath, "{project_id}",
+		getBlackWhiteListClient.ProjectID)
+
+	getBlackWhiteListqueryParams := buildGetBlackWhiteListQueryParams(d)
+	getBlackWhiteListPath += getBlackWhiteListqueryParams
+
+	getBlackWhiteListOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getBlackWhiteListResp, err := getBlackWhiteListClient.Request("GET", getBlackWhiteListPath,
+		&getBlackWhiteListOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving black white list")
+	}
+
+	getBlackWhiteListRespBody, err := utils.FlattenResponse(getBlackWhiteListResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	lists, err := jmespath.Search("data.records", getBlackWhiteListRespBody)
+	if err != nil {
+		return diag.Errorf("error parsing data.records from response= %#v", getBlackWhiteListRespBody)
+	}
+
+	val, ok := lists.([]interface{})
+	if !ok {
+		diag.Errorf("data.records is not a list, data.records= %#v", lists)
+	}
+
+	if len(val) != 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving black white list")
+	}
+
+	list := val[0]
+
+	d.SetId(utils.PathSearch("list_id", list, nil).(string))
```

Comment:
```suggestion
	d.SetId(utils.PathSearch("list_id", list, "").(string))
```

If you want to convert it to string, please do not use **nil**

Reply:
fixed